apps/api/src/provider/minio.provider.ts

```diff
@@ -0,0 +1,74 @@
+import { InternalServerErrorException, Logger, Provider } from '@nestjs/common'
+import * as Minio from 'minio'
+
+export const MINIO_CLIENT = 'MinioClient'
+
+export const MinioProvider: Provider = {
+  provide: MINIO_CLIENT,
+  useFactory: () => {
+    let isServiceLoaded: boolean = false
+
+    const logger = new Logger('MinioProvider')
+    if (
+      !process.env.MINIO_ENDPOINT ||
+      !process.env.MINIO_PORT ||
+      !process.env.MINIO_ACCESS_KEY ||
+      !process.env.MINIO_SECRET_KEY ||
+      !process.env.MINIO_BUCKET_NAME
+    ) {
+      logger.warn(
+        'Missing one or more MinIO environment variables. Skipping initialization...'
+      )
+      return
+    }
+
+    const minioClient: Minio.Client = new Minio.Client({
```

Comment:
Use try-catch in here because it might run into permission errors. If you face any error, make sure the application exits with a non-zero status code. Make sure you emit the error log.

Reply:
noted.